File: php/features.php
<?php
declare(strict_types=1);

// FootballData SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class FootballDataFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new FootballDataBaseFeature();
            case "test":
                return new FootballDataTestFeature();
            default:
                return new FootballDataBaseFeature();
        }
    }
}

File: py/features.py
# FootballData SDK feature factory

from feature.base_feature import FootballDataBaseFeature
from feature.test_feature import FootballDataTestFeature


def _make_feature(name):
    features = {
        "base": lambda: FootballDataBaseFeature(),
        "test": lambda: FootballDataTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
